src/main/java/io/nats/client/impl/PullMessageManager.java

```diff
@@ -162,18 +162,17 @@ protected ManageResult manageStatus(Message msg) {
             case CONFLICT_CODE:
                 // sometimes just a warning
                 String statMsg = status.getMessage();
-                if (statMsg.startsWith("Exceeded Max")
-                    || statMsg.equals(SERVER_SHUTDOWN)
-                    || statMsg.equals(LEADERSHIP_CHANGE)
-                ) {
+                if (statMsg.startsWith(EXCEEDED_MAX_PREFIX) || statMsg.equals(SERVER_SHUTDOWN))
+                {
                     if (raiseStatusWarnings) {
                         conn.executeCallback((c, el) -> el.pullStatusWarning(c, sub, status));
                     }
                     return STATUS_HANDLED;
                 }
 
-                if (statMsg.equals(BATCH_COMPLETED) ||
-                    statMsg.equals(MESSAGE_SIZE_EXCEEDS_MAX_BYTES))
+                if (statMsg.equals(BATCH_COMPLETED)
+                    || statMsg.equals(LEADERSHIP_CHANGE)
+                    || statMsg.equals(MESSAGE_SIZE_EXCEEDS_MAX_BYTES))
```

Comment:
Moved LEADERSHIP_CHANGE to STATUS_TERMINUS (for simplified pull consuming, terminus translates to the pull is done, will need a new one)